src/index.ts

```diff
@@ -55,7 +55,7 @@ const ref = createRef(githubRef)
 
 const s3BucketName = 'sdk.ably.com';
 const sourcePath = path.resolve(core.getInput('sourcePath'));
-const artifactName = core.getInput('artifactName');
+const artifactName = core.getInput('artifactName').trim(); // empty string indicates no value, i.e. artifact name not specified
```

Comment:
`.trim()` is already called on the returned value by default https://github.com/actions/toolkit/blob/main/packages/core/src/core.ts#L143-L155 so no need to add it here.

Reply:
Also, even though it wouldn't change the behaviour it might be more expressive to pass in the default options:
`{ required: false, trimWhitespace: true}`

Reply:
Thanks, both are great observations.

1. I'm going to add a commit for this pull request that removes the call to `trim()` - though I would observe, in my defence, that this isn't clear from the "API Reference" I was looking at ([here](https://github.com/actions/toolkit/tree/main/packages/core#inputsoutputs)) so I could argue I was being defensive 😝.
2. In terms of passing default options to the `getInput()` call, I will do that for all calls to that method in a subsequent pull request. It's interesting to me that, clearly, the value of `required` in `action.yml` does not appear to actually be policed by the GitHub runtime environment. So, perhaps, passing these constraints/requirements in the `getInput()` calls is more than "expressive", at least when it comes to `required` which has a default value of `false` as far as I can see. 🤔 (`trimWhitespace` defaults to `true`, so I would perhaps prefer not to explicitly override it, preferring _convention over configuration_)

Reply:
1. remove in [59e5d89](https://github.com/ably/sdk-upload-action/pull/46/commits/59e5d897c4ddd557b0671687cca80d40a37f3d82)
2. addressed in https://github.com/ably/sdk-upload-action/pull/47

Reply:
Looks good to me, thanks 👍